packages/repo-app-download-sub-generator/src/prompts/prompts.ts

```diff
@@ -0,0 +1,148 @@
+            const defaultSystem = extractDefaultSystem(quickDeployedAppConfig);
+            const filteredSystemQuestion = systemQuestions.prompts.find((p) => p.name === PromptNames.systemSelection);
+
+            if (filteredSystemQuestion) {
+                const choices = (filteredSystemQuestion as ListQuestion<RepoAppDownloadAnswers>).choices;
+
+                if (Array.isArray(choices)) {
+                    const defaultIndex = choices.findIndex((choice: any) => choice.value.system.name === defaultSystem);
+                    filteredSystemQuestion.default = defaultIndex !== -1 ? defaultIndex : undefined;
+                    systemQuestions.prompts = [filteredSystemQuestion];
+                }
+            }
+        }
+
+        let appList: AppIndex = [];
+        const appSelectionPrompt = [
+            {
+                when: async (answers: RepoAppDownloadAnswers): Promise<boolean> => {
+                    if (answers[PromptNames.systemSelection]) {
+                        appList = await fetchAppListForSelectedSystem(
+                            systemQuestions.answers.connectedSystem?.serviceProvider as AbapServiceProvider,
+                            quickDeployedAppConfig?.appId
+                        );
+                    }
+                    return !!systemQuestions.answers.connectedSystem?.serviceProvider;
+                },
+                type: 'list',
+                name: PromptNames.selectedApp,
+                default: () => (quickDeployedAppConfig?.appId ? 0 : undefined),
+                guiOptions: {
+                    mandatory: !!appList.length,
+                    breadcrumb: t('prompts.appSelection.breadcrumb')
+                },
+                message: t('prompts.appSelection.message'),
+                choices: (): { name: string; value: AppInfo }[] => (appList.length ? formatAppChoices(appList) : []),
+                validate: async (answers: AppInfo): Promise<string | boolean | undefined> => {
+                    // Quick deploy config exists but no apps found
+                    if (quickDeployedAppConfig?.appId && appList.length === 0) {
+                        return t('error.quickDeployedAppDownloadErrors.noAppsFound', {
+                            appId: quickDeployedAppConfig.appId
+                        });
+                    }
+
+                    // No apps available at all
+                    if (appList.length === 0) {
+                        return t('prompts.appSelection.noAppsDeployed');
+                    }
+
+                    // Valid app selected, try to download
+                    if (answers?.appId) {
+                        try {
+                            await downloadApp(answers.repoName);
+                            return true;
+                        } catch (error) {
+                            throw new Error(t('error.appDownloadErrors.appDownloadFailure', { error: error.message }));
```

Comment:
Should you just return the `t('error.appDownloadErrors.appDownloadFailure', { error: error.message }` so it's a validation message? Throwing the error will crash the generator